docs/guides/migration.md

```diff
@@ -0,0 +1,315 @@
+│ be removed in the next major version of the provider.
+╵
+╷
+│ Error: Provider produced inconsistent result after apply
+│
+│ When applying changes to aci_bridge_domain.terraform_bd, provider "provider[\"registry.terraform.io/ciscodevnet/aci\"]" produced an
+│ unexpected new value: .relation_fv_rs_bd_to_ep_ret: was cty.StringVal("uni/tn-terraform_tenant/epRPol-ep_pol"), but now
+│ cty.StringVal("uni/tn-common/epRPol-default").
+╵
+```
+
+This issue can be avoided by using the redefined resources with a configurable name attribute or by ensuring that the tDn can be resolved during the apply phase.
+
+## Changes to Child Objects in Configuration
+
+In migrated resources, all child Managed Objects (MOs) are represented in the configuration as a map or a list of maps, rather than single attribute types (such as string, boolean, or integer). This approach ensures the configuration closely resembles the model, allowing for the addition of redefined attributes to a child without causing breaking changes.
+
+## Changed Child Objects in Payloads
+
+In non-migrated resources, child objects were managed via individual REST API requests, which could lead to a large number of REST API calls for a single resource when many children were defined. For migrated resources, the decision is made to include all children inside a single GET/POST request to limit the number of REST API calls.
+
+## Changed Annotation Behavior
+
+The annotation attribute can be set for each MO that exposes the attribute, including children inside a resource. By default, the annotation is set to `orchestrator:terraform`, but this default can be overwritten at the provider level.
+
+```hcl
+provider "aci" {
+  username   = "admin"
+  password   = "password"
+  url        = "https://my-cisco-aci.com"
+  annotation = "provider-level-annotation-overwrite"
+}
+```
+
+## Showing ID in Plan for Create
+
+For non-migrated resources, the ID of the resource appears as "known after apply". In the case of migrated resources, the ID is calculated during the planning phase and included in the plan output.
+
+## Error for Existing MO on Create
+
+Migrated resources offer a provider-level option (`allow_existing_on_create`) to check during the plan if an object already exists. By default, existing MOs can be managed, but this option can be set to `false`, providing a safeguard mechanism to prevent managing the same MO by different configuration parts. The drawback of this safeguard mechanism is an additional API call per resource during the plan to verify that the object does not already exist.
+
+## Allowing Empty Input for Attributes
+
+Terraform supports three states for any value: null (missing), unknown ("known after apply"), and known. Previous SDKs did not support null and unknown states, which prevented differentiating between an empty value and a non-provided value. Because of this limitation, updating a string value to an empty string was not possible before the Plugin Framework.
+
```

Comment:
I would add a clear statement that it is now supported. It currently let the user assume it is supported but we should be clearer.